src/seekcentral.cpp

```diff
@@ -423,28 +434,31 @@ namespace Sleipnir {
             }
 
             if (isFirst) {
-                string err = "Error: no dataset contains any of the query genes";
-                fprintf(stderr, "%s\n", err.c_str());
+                string err = "Error: no dataset contains enough query genes";
+                fprintf(stderr, "%s (%d)\n", err.c_str(), l);
                 if (m_bEnableNetwork) {
                     CSeekNetwork::Send(m_iClient, err);
                 } else if (m_useRPC) {
                     throw query_error(FILELINE + err);
                 }
-                return false;
+                if (m_vecstrAllQuery.size() == 1)
+                  return false;
             }
 
             if (l != m_searchdsetMap.size() - 1) {
                 ss << "|";
             }
 
             //fprintf(stderr, "ss %s\n", ss.str().c_str());
+
+            // iterate over query genes
             isFirst = true;
             for (j = 0; j < m_vecstrAllQuery[l].size(); j++) {
                 sq << m_vecstrAllQuery[l][j] << ":" << count[j];
                 if (j != m_vecstrAllQuery[l].size() - 1) {
                     sq << ";";
                 }
-                if (count[j] == 0) continue;
+                if (count[j] == 0) continue; // only add genes with datasets
```

Comment:
Could this cause size of aq and ss to diverge and trigger this check to fail
'if (sd.size() != qq.size())' on line 521?

Reply:
In principle, it would generate an empty token, but the number of tokens should be the same (i.e. sd.size() and qq.size() would be equal). I added the equality check as a safeguard, though in theory it should never trigger.